src/main.rs

```diff
@@ -61,67 +61,104 @@ fn main() -> Result<(), Box<dyn Error>> {
-                            running_app.move_pose_estimate(0.0, 0.0, -distance);
-                        }
-                        Key::Char('w') => {
-                            running_app.mode = app::AppModes::SendPose;
-                            running_app.move_pose_estimate(distance, 0.0, 0.0);
-                        }
-                        Key::Char('s') => {
-                            running_app.mode = app::AppModes::SendPose;
-                            running_app.move_pose_estimate(-distance, 0.0, 0.0);
-                        }
-                        Key::Char('d') => {
-                            running_app.mode = app::AppModes::SendPose;
-                            running_app.move_pose_estimate(0.0, -distance, 0.0);
-                        }
-                        Key::Char('a') => {
-                            running_app.mode = app::AppModes::SendPose;
-                            running_app.move_pose_estimate(0.0, distance, 0.0);
-                        }
-                        Key::Esc => {
-                            running_app.mode = app::AppModes::RobotView;
-                            // running_app.reset_pose_estimate();
-                        }
-                        Key::Char('\n') => {
-                            initial_pose_pub.send_estimate(&running_app.get_pose_estimate());
-                            running_app.mode = app::AppModes::RobotView;  
-                        }
-                        Key::Char('-') => {
-                            running_app.decrease_zoom();
-                            running_app.compute_bounds(listener.clone());
-                        }
-                        Key::Char('=') => {
-                            running_app.increase_zoom();
-                            running_app.compute_bounds(listener.clone());
-                        }
-                        Key::Ctrl('c') => {
-                            break;
-                        }
-                        Key::Char('k') => {
-                            distance = distance * 2.;
-                        }
-                        Key::Char('j') => {
-                            distance = distance * 0.5;
-                        }
-                        Key::Char('h') => {
-                            running_app.mode = app::AppModes::HelpPage;
-                        }
-                        _ => {}
-                    },
+                    Event::Input(input) => {
+                        match running_app.mode {
+                            app::AppModes::Teleoperate => {
+                                teleoperator.handle_input_from_key(input);
+                            }
+                            _ => (),
+                        }
+                        match input {
+                            Key::Char('q') => {
+                                if running_app.mode != app::AppModes::Teleoperate {
```

Comment:
why not one button to toggle between the modes this would save duplicating one line 6 times

't' springs into my mind

Reply:
Because with more than two modes one would get crazy with cycling. Alternatives:
- fixed numbers: 1 -> view/send pose, 2 -> teleop, 3-> send pose on arbitrary topic (we should think about this), 4 -> 3D first-person view with flight, 5 -> disable safety and take a machine gun, etc.
- a menu: press 'm', a list is visualized, choose with arrows or with numbers as shortcut (personally I like this and would love to implement it)

Reply:
i like the menu, can you create an issue?